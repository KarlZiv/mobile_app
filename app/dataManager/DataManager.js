export default class DataManager {
  static myInstance = null;
  currentUser = {};
  userID = "";
  collectionID = "";
  newLabel = "What Emotion Did You Feel?";
  newCollectionTitle = "Select A Collection";

  users = [
    {
      id: "user2",
      name: "Dwayne Johnson",
      email: "therock@gmail.com",
      password: "rock1234",
      image: require("../assets/theRock.jpeg"),
    },
    {
      id: "user1",
      name: "Cristiano Ronaldo",
      email: "cr7@gmail.com",
      password: "siuu1234",
      image: require("../assets/ronaldo.jpeg"),
    },
  ];

  currentEditMemory = {
    userID: "",
    id: 0,
    collection: "",
    title: "",
    message: "",
    label: "",
    favourite: false,
  };

  memoriesData = [
    {
      userID: "user2",
      id: 1,
      collection: "Personal",
      title: "First day on a set",
      message: "Today was my first day filming my blockbuster film",
      label: "Excited",
      favourite: true,
      date: "22-3-1978",
    },
    {
      userID: "user2",
      id: 2,
      collection: "Personal",
      title: "Business Venture",
      message: "Today I launched my very own brand of Tequila!",
      label: "Happy",
      favourite: true,
      date: "2-9-2018",
    },
    {
      userID: "user2",
      id: 3,
      collection: "Family",
      title: "Divorce",
      message: "We ended our marriage",
      label: "Sad",
      favourite: false,
      date: "12-3-2001",
    },
    {
      userID: "user2",
      id: 4,
      collection: "Family",
      title: "Birth",
      message: "We welcome our first baby into the world",
      label: "Happy",
      favourite: true,
      date: "23-4-2007",
    },
    {
      userID: "user2",
      id: 5,
      collection: "Friends",
      title: "Boys Trip",
      message: "Went to Vegas with my closest pals",
      label: "Excited",
      favourite: false,
      date: "11-11-2021",
    },
    {
      userID: "user2",
      id: 6,
      collection: "Friends",
      title: "Friend",
      message: "Met Vin Diesel, what a cool dude",
      label: "Happy",
      favourite: false,
      date: "1-12-2009",
    },
    {
      userID: "user1",
      id: 7,
      collection: "Friends",
      title: "Football",
      message:
        "Today I played footy with my best friends and I scored the most points (of course)",
      label: "Happy",
      favourite: true,
      date: "13-03-2012",
    },
    {
      userID: "user1",
      id: 8,
      collection: "Family",
      title: "Love",
      message: "I married Georgina",
      label: "Excited",
      favourite: true,
      date: "01-12-2018",
    },
    {
      userID: "user1",
      id: 9,
      collection: "Personal",
      title: "Siuu",
      message: "I yelled SIIUUU, the crowd went wild",
      label: "Happy",
      favourite: true,
      date: "7-4-2019",
    },
    {
      userID: "user1",
      id: 10,
      collection: "Personal",
      title: "Injury",
      message: "Yet another injury, going to rehab",
      label: "Sad",
      favourite: false,
      date: "1-8-2021",
    },
    {
      userID: "user1",
      id: 11,
      collection: "Personal",
      title: "FIFA World Player of the Year",
      message: "All my hard work has paid off!",
      label: "Happy",
      favourite: true,
      date: "18-11-2015",
    },
    {
      userID: "user1",
      id: 12,
      collection: "Family",
      title: "Sons Death",
      message: "My son died during childbirth",
      label: "Sad",
      favourite: false,
      date: "18-4-2022",
    },
  ];

  collectionData = [
    {
      id: 1,
      title: "Friends",
      subtitle: "Memories Shared With Friends",
      image: require("../assets/friends.png"),
    },
    {
      id: 2,
      title: "Family",
      subtitle: "A Collection of Memories With Family",
      image: require("../assets/family.png"),
    },
    {
      id: 3,
      title: "Personal",
      subtitle: "A Personal Collection of Memories",
      image: require("../assets/personal.png"),
    },
  ];

  emotionData = [
    {
      id: 1,
      title: "Happy",
    },
    {
      id: 2,
      title: "Excited",
    },
    {
      id: 3,
      title: "Sad",
    },
  ];

  static geInstance() {
    if (DataManager.myInstance == null) {
      DataManager.myInstance = new DataManager();
    }
    return this.myInstance;
  }

  addMemory(title, messageContent, collection, emotion) {
    // testing for empty, if empty then assigning default values
    if (title === "") {
      title = "--";
    }
    if (messageContent === "") {
      messageContent = "--";
    }
    // testing for placeholder, if placeholder then assign default
    if (collection === "Select A Collection") {
      collection = "Personal";
    }

    if (emotion === "What Emotion Did You Feel?") {
      emotion = "Happy";
    }

    let date = new Date().getDate();
    let month = new Date().getMonth() + 1;
    let year = new Date().getFullYear();
    let seconds = new Date().getSeconds();

    let id = this.memoriesData.length + seconds + 1;

    for (let i = 0; i < this.memoriesData.length; i++) {
      if (id === this.memoriesData[i].id) {
        id = id * 2;
      }
    }

    let memory = {
      userID: this.userID,
      id: id,
      collection: collection,
      title: title,
      message: messageContent,
      label: emotion,
      favourite: false,
      date: date + "-" + month + "-" + year,
    };
    this.memoriesData.push(memory);
  }

  setUserID(id) {
    this.userID = id;
  }

  getUserID() {
    return this.userID;
  }

  getUsersName = () => {
    let user = this.users.filter((user) => user.id === this.userID);
    if (user.length > 0) {
      return user[0].name;
    }
  };

  getUserImage = () => {
    let user = this.users.filter((user) => user.id === this.userID);
    if (user.length > 0) {
      return user[0].image;
    }
  };

  setCollection(inColID) {
    this.collectionID = inColID;
  }

  getCollectionTitle() {
    let collection = this.collectionData.filter(
      (collection) => collection.id === this.collectionID
    );

    if (collection.length > 0) {
      return collection[0].title;
    }
  }

  getUserMemories() {
    let memories = this.memoriesData.filter(
      (memory) => memory.userID === this.userID
    );
    if (memories.length > 0) {
      return memories;
    }
  }

  getFavouriteMemories() {
    let memories = this.memoriesData.filter(
      (memory) => memory.userID === this.userID && memory.favourite === true
    );
    if (memories.length > 0) {
      return memories;
    }
  }

  getCollectionMemories() {
    let memories = this.memoriesData.filter(
      (memory) =>
        memory.collection === this.getCollectionTitle() &&
        memory.userID === this.userID
    );
    if (memories.length > 0) {
      return memories;
    }
  }

  setNewCollectionTitle(inNewTitle) {
    this.newCollectionTitle = inNewTitle;
  }

  getNewCollectionTitle() {
    return this.newCollectionTitle;
  }
  setNewLabel(inNewLabel) {
    this.newLabel = inNewLabel;
  }

  getNewLabel() {
    return this.newLabel;
  }

  getMemoryDate(id) {
    for (let i = 0; i < this.memoriesData.length; i++) {
      if (this.memoriesData[i].id === id) {
        return this.memoriesData[i].date;
      }
    }
  }

  getFilteredEmotions() {
    if (this.newLabel !== "What Emotion Did You Feel?") {
      let memories = this.memoriesData.filter(
        (memory) =>
          memory.label === this.newLabel &&
          memory.userID === this.userID &&
          memory.collection === this.getCollectionTitle()
      );
      if (memories.length > 0) {
        return memories;
      }
    } else {
      return this.getCollectionMemories();
    }
  }

  setPrefills(userID, id, collection, title, message, label) {
    this.currentEditMemory.userID = userID;
    this.currentEditMemory.id = id;
    this.currentEditMemory.collection = collection;
    this.currentEditMemory.title = title;
    this.currentEditMemory.message = message;
    this.currentEditMemory.label = label;
  }

  getCollectionPrefill() {
    return this.currentEditMemory.collection;
  }
  getTitlePrefill() {
    return this.currentEditMemory.title;
  }
  getMemoryContentsPrefill() {
    return this.currentEditMemory.message;
  }
  getEmotionPrefill() {
    return this.currentEditMemory.label;
  }

  delete(memory) {
    this.memoriesData = this.memoriesData.filter(
      (item) => item.id !== memory.id
    );
  }

  updateMemories(id, title, message, collection, emotion) {
    for (let i = 0; i < this.memoriesData.length; i++) {
      if (this.memoriesData[i].id === id) {
        if (!(this.memoriesData[i].title === title)) {
          this.memoriesData[i].title = title;
        }
        if (!(this.memoriesData[i].message === message)) {
          this.memoriesData[i].message = message;
        }
        if (!(this.memoriesData[i].collection === collection)) {
          if (this.newCollectionTitle !== "Select A Collection") {
            this.memoriesData[i].collection = collection;
            this.setNewCollectionTitle("Select A Collection");
          }
        }
        if (!(this.memoriesData[i].emotion === emotion)) {
          if (this.newLabel !== "What Emotion Did You Feel?") {
            this.memoriesData[i].label = emotion;
            this.setNewLabel("What Emotion Did You Feel?");
          }
        }
        break;
      }
    }
  }

  addUser(name, email, password) {
    let id = "user" + this.users.length + 1;
    let user = {
      id: id,
      name: name,
      email: email,
      password: password,
      image: require("../assets/placeholder.jpeg"),
    };
    this.users.push(user);
  }
}
