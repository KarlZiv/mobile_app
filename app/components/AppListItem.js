import React from "react";
import { View, StyleSheet, TouchableHighlight } from "react-native";
import { Swipeable } from "react-native-gesture-handler";
import { MaterialCommunityIcons } from "@expo/vector-icons";

import AppColors from "../config/AppColors";
import AppText from "./AppText";
import DataManager from "../dataManager/DataManager";

let commonData = DataManager.geInstance();

function AppListItem({
  title,
  message,
  style,
  underlayColor,
  onSwipeLeft,
  icon,
  id,
}) {
  return (
    <Swipeable renderRightActions={onSwipeLeft}>
      <TouchableHighlight underlayColor={underlayColor}>
        <View style={[styles.container, style]}>
          <View style={styles.view}>
            <View style={styles.titleContainer}>
              <View style={styles.titleContainerInner}>
                <AppText style={styles.itemTitle}>{title}</AppText>
              </View>
              <View style={styles.iconContainerInner}>
                <MaterialCommunityIcons
                  style={styles.icon}
                  name={icon}
                  size={26}
                  color={AppColors.otherColor}
                />
              </View>
            </View>
            <AppText style={styles.itemDate}>
              {commonData.getMemoryDate(id)}
            </AppText>
            {message && <AppText style={styles.itemMessage}>{message}</AppText>}
          </View>
        </View>
      </TouchableHighlight>
    </Swipeable>
  );
}

const styles = StyleSheet.create({
  container: {
    flexDirection: "column",
    backgroundColor: AppColors.tertiaryColor,
    height: 120,
    width: "100%",
  },
  view: {
    marginLeft: 10,
  },
  itemTitle: {
    color: AppColors.otherColor,
    marginTop: 15,
    padding: 2,
    fontSize: 16,
  },
  itemDate: {
    color: AppColors.secondaryColor,
    fontSize: 12,
    padding: 2,
  },
  itemMessage: {
    color: AppColors.secondaryColor,
    fontSize: 14,
    marginTop: 2,
    padding: 2,
    width: "75%",
  },
  titleContainer: {
    flexDirection: "row",
  },
  icon: { marginRight: 20, top: 5 },
  titleContainerInner: {
    width: "80%",
  },
  iconContainerInner: {
    width: "20%",
    justifyContent: "center",
    alignItems: "flex-end",
  },
});

export default AppListItem;
