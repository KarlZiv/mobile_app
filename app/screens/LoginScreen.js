import React from "react";
import { StyleSheet, StatusBar } from "react-native";
import { Formik } from "formik";
import * as yup from "yup";

import AppTextInput from "../components/AppTextInput";
import AppColors from "../config/AppColors";
import AppScreen from "../components/AppScreen";
import AppText from "../components/AppText";
import AppLogo from "../components/AppLogo";
import AppButton from "../components/AppButton";
import DataManager from "../dataManager/DataManager";

let commonData = DataManager.geInstance();
const users = commonData.users;

const schema = yup.object().shape({
  email: yup.string().required().email().label("Email"),
  password: yup.string().required().min(8).max(14).label("Password"),
});

const validateUser = (email, password) => {
  return (
    users.filter((user) => user.email === email && user.password === password)
      .length > 0
  );
};

const getUser = ({ email }) => {
  return users.find((user) => user.email === email);
};

const createUser = ({ email }) => {
  let userID = getUser({ email }).id;
  commonData.setUserID(userID);
};

function LoginScreen({ navigation }) {
  StatusBar.setBarStyle("dark-content", true);
  return (
    <AppScreen style={styles.container}>
      <AppLogo style={styles.logo} />
      <Formik
        initialValues={{ email: "", password: "" }}
        onSubmit={(values, { resetForm }) => {
          if (validateUser(values.email, values.password)) {
            resetForm();
            createUser(values);
            navigation.navigate("Home");
          } else {
            resetForm();
            alert("Invalid Login Details.");
          }
        }}
        validationSchema={schema}
      >
        {({ handleChange, handleSubmit, errors, setFieldTouched, touched }) => (
          <>
            <AppTextInput
              icon="email"
              placeholder="Email"
              autoCapitalize="none"
              onChangeText={handleChange("email")}
              onBlur={() => setFieldTouched("email")}
              style={styles.textInput}
              id="userEmail"
              testID="loginEmail"
            />
            {touched.email && <AppText>{errors.email}</AppText>}
            <AppTextInput
              icon="lock"
              placeholder="Password"
              secure={true}
              onChangeText={handleChange("password")}
              onBlur={() => setFieldTouched("password")}
              style={styles.textInput}
            />
            {touched.password && <AppText>{errors.password}</AppText>}

            <AppButton
              title="Login"
              onPress={() => {
                handleSubmit();
              }}
              style={styles.button}
            />
          </>
        )}
      </Formik>
      <AppButton
        style={styles.outerButton}
        title={"Back"}
        onPress={() => {
          navigation.navigate("Welcome");
        }}
      />
    </AppScreen>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: AppColors.primaryColor,
    alignItems: "center",
    justifyContent: "center",
  },
  form: {},
  logo: {
    marginTop: -150,
  },
  button: {
    marginTop: 30,
    alignSelf: "center",
  },
  outerButton: {
    position: "absolute",
    width: 80,
    left: 10,
    top: 50,
  },
  text: {
    fontSize: 18,
    marginBottom: 10,
    zIndex: 100,
  },
  textInput: {
    width: "80%",
  },
});

export default LoginScreen;
