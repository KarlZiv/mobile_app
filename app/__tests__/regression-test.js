import React from "react";
import renderer from "react-test-renderer";
import AppListItem from "../components/AppListItem";
import DataManager from "../dataManager/DataManager";
import LoginScreen from "../screens/LoginScreen";

jest.useFakeTimers();

let commonData = DataManager.geInstance();

test("Adding a memory with placeholder/default values", () => {
  commonData.setUserID("user1");
  commonData.addMemory(
    "",
    "",
    "Select A Collection",
    "What Emotion Did You Feel?"
  );
  let addedMemory = commonData.getUserMemories().pop();
  expect(addedMemory.title).toBe("--");
  expect(addedMemory.collection).toBe("Personal");
  const rendered = renderer
    .create(
      <AppListItem
        title={addedMemory.title}
        message={addedMemory.message}
        id={addedMemory.id}
        icon={"emoticon-" + addedMemory.label.toLowerCase()}
      />
    )
    .toJSON();
  expect(rendered).toMatchSnapshot();
});

test("Adding a memory with valid user inputs", () => {
  commonData.setUserID("user1");
  commonData.addMemory("Taco Tuesday", "Tacos on Tuesday", "Personal", "Happy");
  let addedMemory = commonData.getUserMemories().pop();
  expect(addedMemory.title).toBe("Taco Tuesday");
  expect(addedMemory.collection).toBe("Personal");
  const rendered = renderer
    .create(
      <AppListItem
        title={addedMemory.title}
        message={addedMemory.message}
        id={addedMemory.id}
        icon={"emoticon-" + addedMemory.label.toLowerCase()}
      />
    )
    .toJSON();
  expect(rendered).toMatchSnapshot();
});

test("Display all user inputs required for user to login", () => {
  const tree = renderer.create(<LoginScreen />).toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree.children[1].children[1].props.placeholder).toBe("Email");
  expect(tree.children[2].children[1].props.placeholder).toBe("Password");
});

// // testing if company form is working
// describe("User Login tests ", () => {
//   it("Success test", async () => {
//     const handleSubmit = jest.fn();

//     const LoginScreen = render(<LoginScreen />);
//     const form = LoginScreen.getByTestId("form");

//     const email = form.getByTestId("loginEmail");
//     const password = form.getByTestId("loginPassword");
//     console.log(email);

//     // const submitButton = LoginScreen.getByTestId("companyFormSubmitButton");

//     // userEvent.type(email, "therock@gmail.com");
//     // userEvent.type(password, "rock1234");
//     // userEvent.click(submitButton);
//     // await waitFor(() =>
//     //   expect(handleSubmit).toHaveBeenCalledWith({
//     //     password: "rock1234",
//     //     email: "therock@gmail.com",
//     //   })
//     // );
//   });
// });

// describe("Logging into Application", () => {
//   it("Should display all user inputs required for user to login", () => {
//     const tree = renderer.create(<LoginScreen />).toJSON();
//     expect(tree).toMatchSnapshot();
//   });

// expect(
//   instanceOfLoginScreen.validateUser("therock@gmail.com", "rock12345678")
// ).tobe(true);
//});

// Enzyme.configure({ adapter: new Adapter() });
// describe("LoginScreen", () => {
//   it("should show user input", () => {
//     const wrapper = shallow(<LoginScreen />);
//     const appScreen = wrapper.find(AppScreen);
//     const formik = appScreen.find(Formik);
//     const userTextInput = appScreen.find(AppTextInput);

//     // const mockEvent = {
//     //   target: { name: "email", value: "therock@gmail.com" },
//     // };
//     // const userTextInput = userInput.find(Formik).dive().props().children.props
//     //   .children[0];

//     //const propsOfUserTextInput = userTextInput.length;

//     //userTextInput.simulate("onChange", mockEvent);

//     // obtaining values.
//     //const userInput2 = userInput.find(Formik).dive().props().value.values;

//     console.log(appScreen.length);
//   });
//   // it("should hide text when button is clicked", () => {
//   //   const wrapper = shallow(<LoginScreen />);
//   //   const button = wrapper.find(<AppButton />);
//   //   button.simulate("click");
//   // });
// });

// describe("LoginScreen", () => {
//   test("should update email field on change", () => {
//     const tree = shallow(<LoginScreen />);
//     const appScreen = tree.find(AppScreen);
//     const emailInput = appScreen.findWhere(
//       (node) => node.prop("testID") === "loginEmail"
//     );
//     console.log(emailInput.props());
//     emailInput
//       .simulate("change", {
//         persist: () => {},
//         target: {
//           name: "email",
//           value: "email@gmail.com",
//         },
//       })
//       .update();
//     expect(emailInput.html()).toMatch("email@gmail.com");
//   });
// });
