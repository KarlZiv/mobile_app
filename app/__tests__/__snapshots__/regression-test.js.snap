// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Adding a memory with placeholder/default values 1`] = `
<View
  collapsable={false}
  forwardedRef={[Function]}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  onLayout={[Function]}
  style={
    Object {
      "overflow": "hidden",
    }
  }
>
  <View
    collapsable={false}
    forwardedRef={[Function]}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    pointerEvents="auto"
    style={
      Object {
        "transform": Array [
          Object {
            "translateX": -0,
          },
        ],
      }
    }
  >
    <View
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#252422",
              "flexDirection": "column",
              "height": 120,
              "width": "100%",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Object {
              "marginLeft": 10,
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Object {
                  "width": "80%",
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#252422",
                      "fontFamily": "Avenir-Roman",
                      "fontSize": 16,
                      "fontWeight": "bold",
                    },
                    Object {
                      "color": "#EB5E28",
                      "fontSize": 16,
                      "marginTop": 15,
                      "padding": 2,
                    },
                  ]
                }
              >
                --
              </Text>
            </View>
            <View
              style={
                Object {
                  "alignItems": "flex-end",
                  "justifyContent": "center",
                  "width": "20%",
                }
              }
            >
              <Text />
            </View>
          </View>
          <Text
            style={
              Array [
                Object {
                  "color": "#252422",
                  "fontFamily": "Avenir-Roman",
                  "fontSize": 16,
                  "fontWeight": "bold",
                },
                Object {
                  "color": "#CCC5B9",
                  "fontSize": 12,
                  "padding": 2,
                },
              ]
            }
          >
            21-5-2022
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#252422",
                  "fontFamily": "Avenir-Roman",
                  "fontSize": 16,
                  "fontWeight": "bold",
                },
                Object {
                  "color": "#CCC5B9",
                  "fontSize": 14,
                  "marginTop": 2,
                  "padding": 2,
                  "width": "75%",
                },
              ]
            }
          >
            --
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`Adding a memory with valid user inputs 1`] = `
<View
  collapsable={false}
  forwardedRef={[Function]}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  onLayout={[Function]}
  style={
    Object {
      "overflow": "hidden",
    }
  }
>
  <View
    collapsable={false}
    forwardedRef={[Function]}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    pointerEvents="auto"
    style={
      Object {
        "transform": Array [
          Object {
            "translateX": -0,
          },
        ],
      }
    }
  >
    <View
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#252422",
              "flexDirection": "column",
              "height": 120,
              "width": "100%",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Object {
              "marginLeft": 10,
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Object {
                  "width": "80%",
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#252422",
                      "fontFamily": "Avenir-Roman",
                      "fontSize": 16,
                      "fontWeight": "bold",
                    },
                    Object {
                      "color": "#EB5E28",
                      "fontSize": 16,
                      "marginTop": 15,
                      "padding": 2,
                    },
                  ]
                }
              >
                Taco Tuesday
              </Text>
            </View>
            <View
              style={
                Object {
                  "alignItems": "flex-end",
                  "justifyContent": "center",
                  "width": "20%",
                }
              }
            >
              <Text />
            </View>
          </View>
          <Text
            style={
              Array [
                Object {
                  "color": "#252422",
                  "fontFamily": "Avenir-Roman",
                  "fontSize": 16,
                  "fontWeight": "bold",
                },
                Object {
                  "color": "#CCC5B9",
                  "fontSize": 12,
                  "padding": 2,
                },
              ]
            }
          >
            21-5-2022
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#252422",
                  "fontFamily": "Avenir-Roman",
                  "fontSize": 16,
                  "fontWeight": "bold",
                },
                Object {
                  "color": "#CCC5B9",
                  "fontSize": 14,
                  "marginTop": 2,
                  "padding": 2,
                  "width": "75%",
                },
              ]
            }
          >
            Tacos on Tuesday
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`Display all user inputs required for user to login 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "alignContent": "center",
        "display": "flex",
        "flex": 1,
        "marginTop": 0,
      },
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFCF2",
        "flex": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "marginBottom": 60,
          "marginTop": 100,
        },
        Object {
          "marginTop": -150,
        },
      ]
    }
  >
    <Image
      source={1}
      style={
        Object {
          "height": 120,
          "width": 120,
        }
      }
    />
    <Text
      style={
        Array [
          Object {
            "color": "#252422",
            "fontFamily": "Avenir-Roman",
            "fontSize": 16,
            "fontWeight": "bold",
          },
          Object {
            "fontSize": 20,
            "fontWeight": "900",
          },
        ]
      }
    >
      MyMems
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#252422",
          "flexDirection": "row",
          "height": 50,
          "marginVertical": 5,
          "width": 250,
        },
        Object {
          "width": "80%",
        },
      ]
    }
  >
    <Text />
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      autoCorrect={false}
      id="userEmail"
      maxLength={35}
      onBlur={[Function]}
      onChangeText={[Function]}
      placeholder="Email"
      placeholderTextColor="#CCC5B9"
      rejectResponderTermination={true}
      style={
        Object {
          "color": "#CCC5B9",
          "fontFamily": "Avenir Next",
          "fontSize": 18,
          "width": "80%",
        }
      }
      testID="loginEmail"
      underlineColorAndroid="transparent"
    />
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#252422",
          "flexDirection": "row",
          "height": 50,
          "marginVertical": 5,
          "width": 250,
        },
        Object {
          "width": "80%",
        },
      ]
    }
  >
    <Text />
    <TextInput
      allowFontScaling={true}
      autoCorrect={false}
      maxLength={35}
      onBlur={[Function]}
      onChangeText={[Function]}
      placeholder="Password"
      placeholderTextColor="#CCC5B9"
      rejectResponderTermination={true}
      secureTextEntry={true}
      style={
        Object {
          "color": "#CCC5B9",
          "fontFamily": "Avenir Next",
          "fontSize": 18,
          "width": "80%",
        }
      }
      underlineColorAndroid="transparent"
    />
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "#EB5E28",
        "borderRadius": 50,
        "elevation": 6,
        "marginTop": 30,
        "opacity": 1,
        "padding": 10,
        "shadowColor": "rgba(0, 0, 0, 0.3)",
        "shadowOffset": Object {
          "height": 5,
          "width": 1,
        },
        "shadowOpacity": 0.8,
        "shadowRadius": 5,
        "width": 180,
      }
    }
  >
    <View>
      <Text
        style={
          Array [
            Object {
              "color": "#252422",
              "fontFamily": "Avenir-Roman",
              "fontSize": 16,
              "fontWeight": "bold",
            },
            Object {
              "color": undefined,
              "fontSize": undefined,
            },
          ]
        }
      >
        Login
      </Text>
    </View>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#EB5E28",
        "borderRadius": 50,
        "elevation": 6,
        "left": 10,
        "opacity": 1,
        "padding": 10,
        "position": "absolute",
        "shadowColor": "rgba(0, 0, 0, 0.3)",
        "shadowOffset": Object {
          "height": 5,
          "width": 1,
        },
        "shadowOpacity": 0.8,
        "shadowRadius": 5,
        "top": 50,
        "width": 80,
      }
    }
  >
    <View>
      <Text
        style={
          Array [
            Object {
              "color": "#252422",
              "fontFamily": "Avenir-Roman",
              "fontSize": 16,
              "fontWeight": "bold",
            },
            Object {
              "color": undefined,
              "fontSize": undefined,
            },
          ]
        }
      >
        Back
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
